ingest/scripts/call_loculus.py

```diff
@@ -171,6 +173,65 @@ def get_or_create_group_and_return_group_id(config: Config, allow_creation: bool
     return create_group_and_return_group_id(config)
 
 
+def filter_fasta_by_submission_ids(
+    fasta_dict: dict[str, Any], submission_ids: list[str], config: Config
+):
+    if config.segmented:
+        segmented_submission_ids = [
+            f"{submission_id}_{segment}"
+            for submission_id in submission_ids
+            for segment in config.nucleotide_sequences
+        ]
+        return [
+            fasta_dict[submission_id]
+            for submission_id in segmented_submission_ids
+            if submission_id in fasta_dict
+        ]
+    return [
+        fasta_dict[submission_id] for submission_id in submission_ids if submission_id in fasta_dict
+    ]
+
+
+def post_fasta_batches(
+    url,
+    fasta_file: str,
+    metadata_file: str,
```

Comment:
I _think_ the types here might be off? In line 273 you pass in a file pointer not a file name, so it's not `str` at least for the `fasta_file`. If `SeqIO.parse` can read a filename, you might be able to just ditch the `open(..)` in line 273 and instead pass in file names?